### DISCUSSION TITLE
Summer Internship Project Ideas

### DISCUSSION THREAD

**Discussion by ChristopherA** [BlockchainCommons/Community#76] [Ideas]
This summer we will be having our third cohort of summer interns participating in our virtual internship program (see: https://github.com/BlockchainCommons/Community/discussions/74):

What makes for a good internship project?

* It should be something small in scope (~40 hours over two months per intern involved) but with a clear deliverable. 
* The project should involve the talents of more than one intern (we want all projects to be group projects).
* Not all interns will have solid developer experience, so a project should contain other tasks, such as a user survey, use cases, requirements, testing, docs, website, etc.
* Should make sense from the perspective of the [vision & mission](https://www.blockchaincommons.com/vision.html) of Blockchain Commons.
* Should be a good learning experience for the interns.

As the Human Rights Foundation @HRF is specifically funding some of the internships, I'd also love to see some projects that include human rights goals, such as teach advocates how to leverage our solutions ,or identify specific use cases that our ecosystem is not supporting. Many of these may not require software engineering skills.

Also let us know if you'd be interested in mentoring the internship team if they choose to work on this project.

-- Christopher Allen

EDIT: If you choose a project to work on, please ASSIGN yourself to the associated Issue, which should be linked from the Project description below.

**Comment by ChristopherA** [comment]
## Project: Adapt LetheKit for SeedSigner

[LetheKit v0](https://github.com/BlockchainCommons/lethekit) is an open source DIY hardware box with offline cryptographic tools. It can use dice for entropy, import seeds in a variety formats, and exports derived keys using UR-based QR codes.

However, it can't support crypto-request QRs, or sign transactions as it does not have a camera. 

[SeedSigner](https://github.com/seedsigner/seedsigner/) is another open source DIY device that does have a camera and a better display but it does not support UR QRs. 

The goal would be to evaluate if we can submit patches to @seedsigner to add all the UR functionality that LetheKit has, and if not, port the LetheKit code to that hardware as an alternative to the seedsigner code base.

Sub-projects could be to understand requirements (for instance, LetheKit forgets when powered off, I'm not sure with seedsigner) and use cases, evaluate their code, update the existing LetheKit v0 code for recent UR features, write docs, create a video tutorial, update the LetheKit web page, etc.

@ksedgwic or @gorazdko, fomer leads on LetheKit project might be persuaded to be mentors.

(see issue https://github.com/BlockchainCommons/Community/issues/89)

**Reply by shannona** [reply]
This is available as issue #89.

**Comment by simonratner** [comment]
## Project: NFC-A tag test platform

This project is for interns with interest in hardware, and some experience with hardware development / toolkits.

Adapt an MCU devkit for use as an NFC-A tag test platform for testing [Seed Tool](https://github.com/BlockchainCommons/GordianSeedTool-iOS) NFC read/write functionality.

For example, using:
* [nRF52832DK](https://www.nordicsemi.com/Products/Development-hardware/nRF52-DK) or [nRF52480DK](https://www.nordicsemi.com/Products/Development-hardware/nRF52840-DK)
* Nordic NFC libraries (https://github.com/nrfconnect/sdk-nrfxlib/tree/main/nfc)
* Nordic BLE SoftDevice, or an open OS and BLE stack like Mynewt/Nimble (https://mynewt.apache.org) or Zephyr

It is possible to set up a board that can be configured to emulate an arbitrary NFC-A tag, readable or writable. 
Other combinations of hardware/software are possible - any MCU that has an integrated NFC-A tag peripheral will do.

A mobile app can be used to configure this tag via BLE commands, including loading test data:
* a generic app like [Nordic nRF Connect](https://www.nordicsemi.com/Products/Development-tools/nrf-connect-for-mobile)
* a custom app built for this purpose
* a part of "developer test features" included in [Seed Tool](https://github.com/BlockchainCommons/GordianSeedTool-iOS)

**Reply by simonratner** [reply]
May also be possible with some Adafruit hardware and their existing libraries, although there are some gotchas:
https://forums.adafruit.com/viewtopic.php?f=58&t=180116&p=876623&hilit=nrf52+nfc#p876514

**Reply by shannona** [reply]
This is available as #117.

**Comment by ChristopherA** [comment]
## Project: Inventory of Digital Identity & Blockchain Laws

A good project for non-engineering interns, in particular those with library science or pre-law eduction.

Discover and compare different US state & international laws in regards to digital signatures, digital identity, and blockchain-related technologies, and create an open source repository with catealogue.

* What jurisdictions already have laws or regulations in place? Where can we find the official version?
    * Digital signatures
    * Use of digital records on the blockchain
    * Cryptocurrency related laws
    * IP (copyright, trademark, etc.) on blockchain (not just NFTs)
    * DAO-related
    * Digital Privacy & Digital Human Rights related (in particular, digital freedom of speech, right to assemble, and right to transaction).
    * etc.
* What jurisdictions are in progress on various laws? Who is involved, when are hearings being held, where are the drafts?
* Who is offering model legislation? Advocating for changes?
* What are the various definitions? How are they different? Are some better than others?
* Where are laws or regulations (or proposals) risking locking in various solutions? 
* Who is doing more innovative work in different areas?
* etc.

We have some practicing lawyers that are likely to be able to mentor on this project, and there are opportunities to influence legislation in Wyoming, Texas, Argentina, and more in the coming year.

**Reply by RezaNourmohammadi** [reply]
Hi Christopher, how we can participate in this project?

**Reply by ChristopherA** [reply]
@RezaNourmohammadi at least 3 of our interns for this summer are interested in this project. How would you like to be involved?

**Reply by RezaNourmohammadi** [reply]
It's up to you, I am working on this topic, and I would like to participate in this project. Please let me know what should I do and what would be the process?

**Reply by JRJunior221** [reply]
Collin Kinniry

**Reply by shannona** [reply]
I've flagged this as issue #114

**Comment by ChristopherA** [comment]
## Project: Contribute to Taproot/Schnorr Update of "Learning Bitcoin from the Command Line"

Our popular online course [Learning Bitcoin from the Command Line](https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line) is complete through pre-Taproot/Schnorr release of Bitcoin-Core, but needs research and first drafts on a number of topics:

A list of the items being considered for the next release (3.0?) of the course are described at [Todo List for v3.0 of Learning Bitcoin](https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/blob/master/TODO-30.md).

Note that many of these items do not require a deep knowledge of software engineering, and can be accomplished by student willing to dive into the command-line.

There are many opportunities for mentors here, depending on the sub-topic.

**Reply by shannona** [reply]
The first step here is going to be figuring out how much of the new Schnorr/Taproot functionality is accessible from the command line right now. (But there's also lots of other stuff in that TODO that could be taken on as projects.)

**Reply by kallewoof** [reply]
FWIW btcdeb has a 'tap' binary now, with a tutorial describing usage here: https://github.com/bitcoin-core/btcdeb/blob/master/doc/tapscript-example-with-tap.md

**Reply by ChristopherA** [reply]
Related are updates for latest bitcoin-core https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/issues/575 which probably should be done first.

**Reply by devesh-002** [reply]
Interested in a few projects including this, I have mailed you regarding the same.

**Reply by shannona** [reply]
I've flagged this as issue #115

**Reply by tikka01** [reply]
I am interested in working on this project!

**Comment by ChristopherA** [comment]
## Project: Video Course for "Learning Bitcoin from the Command Line"

Our popular online course [Learning Bitcoin from the Command Line](https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line) is completely text, but there is an opportunity to also teach it in a guided video course.

Not all topics need to be covered, but even a better "intro" video course would be helpful. 

This project would be great for someone new to Bitcoin but comes to it with some video editing experience or background in education.

**Reply by ConorOkus** [reply]
I gave this a go a few years ago here - https://www.youtube.com/playlist?list=PLkS2vJuX8dG4OpmGlUAg0o6VOQJop0bxz

Probably out of date now though...

Great course

**Reply by ChristopherA** [reply]
Cool! I didn't know about this @ConorOkus! I'd certainly have shared it further if knew back then.

Would be you interested in being consulting or being involved if this summer interns choose to tackle this video project again?

I'm also interested in investigating using some of the terminal output "animated screenshot" tools like https://asciinema.org, as they may be more maintainable.

**Reply by ConorOkus** [reply]
Yes sure, I think it would be great to add it to the Bitcoin Developers Youtube channel I have been adding content to - https://www.youtube.com/channel/UCUq_ZdezVWKPvkWRicAYxLA/videos

**Reply by shannona** [reply]
I've recorded this as #119.

**Comment by ChristopherA** [comment]
## Project: Edit "Blockstream Office Hours" Videos

We have slightly more than a dozen Zoom recordings of various Bitcoin, blockchain, and privacy advocates from our "Office Hours" during the last two cohorts of interns. There will be more interviews and Q&A added this summer.

I'd like to see these edited into a series of videos on different kinds topics, for instance, one collating all the blockchain career advice that different people offered into a single video.

There is also an opportunity for us to add more interviews of blockchain luminaries to supplement these videos.

**Reply by shannona** [reply]
This is flagged as #120

**Comment by ChristopherA** [comment]
## Project: Modular Bitcoin-Standup-Scripts

There is a PR  to our Bitcoin Standup scripts from former internet @jodobear that modularizes these scripts to allow for additional open source bitcoin libraries to be chosen to install: https://github.com/BlockchainCommons/Bitcoin-Standup-Scripts/pull/15

However, this project got stalled, and also needs to address some changes in later versions of the release Bitcoin-Standup-Scripts.

I'd also like to see a version of tested against older versions of macOS (in particular Mojave) as the Gordian Server app doesn't work with these old releases, and there are some good reasons why to support Bitcoin-Core on old hardware (in particular Mac minis).

**Reply by shannona** [reply]
We probably had a working modular Bitcoin-Standup Script two years ago, but we let other priorities get in our way, and by the time we went back to the modular scripts they were out of date due to updates to the core script. So, the core idea is to figure out how to merge @jodobear's already great work with the repo as it stands now.

Beyond that, questions include:
* How can we create an interface that maximizes usability? 
* Are there newer modules that we should add?
* How can we document why people would want to use (or not) all of these options?

**Reply by shannona** [reply]
Logged as #121

**Comment by ChristopherA** [comment]
## Project: Human Rights Use Cases for Digital Privacy & Cryptocurrency

This is a great project for non-development interns, but also allows them to connect these topics to software engineering practices.

Write up a number of human-rights centric use cases for privacy and cryptocurrency, including balancing freedom of speech, freedom to assemble, right to work, freedom to transact vs. preventing bad actors from taking advantage of those with less power.

This includes interviewing various major stakeholders EFF, HRF, Red Cross, UNHR, plus people actually in the field doing the work.

Choose one more more use cases to build a 12-step engagement model, similar to Joram or Amira, but focused on human rights and cross-border volunteer workers. Possibly include more detailed risk modeling, threat analysis, etc.

Disseminate the final materials with supporting documents on Blockchain Common’s public website or elsewhere.

Mentors may include Alex Gladstein of HRF, Joe Andrieu, and a number of others.

Related: https://github.com/BlockchainCommons/Community/discussions/61

**Reply by shannona** [reply]
The core idea here is figuring out what might be needed to really support and improve human rights with our blockchain/cryptocurrency/digital-identity technologies. As Christopher says, the best way to figure that out is by interviewing stakeholders. A lot of work in this space has the problem of being handed top-down: the developers decide what's needed and try to convince people why it's important. Something instead coming bottom-up might be much more successful.

Use case articles created by Rebooting the Web of Trust show some initial work on this topic. Search for "Use Cases" on this page:
https://www.weboftrust.info/papers.html

It includes the Amira and Joram use cases that Christopher mentions, and Joe Andrieu's 15-step engagement model.

Amira: https://github.com/WebOfTrustInfo/rwot5-boston/blob/master/final-documents/amira.pdf
Joram: https://github.com/WebOfTrustInfo/rwot3-sf/blob/master/final-documents/joram-engagement-model.pdf

There's a good explanation and chart of the model on the last page or two of each of those papers.

**Reply by JRJunior221** [reply]
Collin Kinniry

**Reply by aisha-ahmadi** [reply]
Aisha Ahmadi

**Reply by shannona** [reply]
Logged as #122

**Reply by tikka01** [reply]
Chaitika Bhatnagar

**Reply by sunintb** [reply]
Interested - happy to collaborate w/ other interns to conduct user interviews and publish an engagement model

**Comment by ChristopherA** [comment]
## Project: Survey of Developer & Blockchain Ethics

Another project that can be led by a non-engineer. I'd like a good survey of history of software engineering ethics and more recent thoughts given blockchain development, self-sovereign identity, modern privacy technologies, etc.

The team should be prepared to be able to lead a group discussion with Blockchain Commons stakeholders (patrons, contributors, volunteers, staff and users) on this topic, and present an article on the topic in our blog.

**Reply by shannona** [reply]
Logged as #123

**Comment by ChristopherA** [comment]
## Project: C-Lightning: Course Update & Maintenance Documents

Our popular online course [Learning Bitcoin from the Command Line](https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line) has some basic chapters on C-Lightning, but we need more:

* Edit for current C-Lighting, but also common plugins and developer tools.
* Consider a fork of the focused on installation and maintenance of a Bitcoin-LN node, that is CLI-focused (rather than GUI-focused) but not coding focused. Should include installation of common admin tools like CLBOSS.
* Define best practices of for keeping LN services private and leverage tor for web services. Currently the common practices is for LN services to be offered on the public networks which is a privacy risk.

This project does not require coding experience, but one of the team should be able to do some basic CLI scripting.

**Reply by shannona** [reply]
Recorded as Issue #124.

**Comment by ChristopherA** [comment]
## Project: Stand-alone BIP-322 (Generic Signed Message Format Leveraging Bitcoin Script) CLI app

This is a project for some more advanced coders with C++ experience.

[BIP-322](https://github.com/bitcoin/bips/blob/master/bip-0322.mediawiki) is a proposal by @kallewoof that extends the old legacy single-key signing of messages, to sign or partially sign a message for any Bitcoin script from which they could conceivably spend.

There is a [PR](https://github.com/bitcoin/bitcoin/pull/16440) to update bitcoin-core that has been languishing for some time, but it should be possible to leverage code from @kallewoof's [Bitcoin Script Debugger](https://github.com/kallewoof/btcdeb) to make a standalone tool that can sign messages.

Potentially @kallewoof might be up for mentoring interns tackling this project.

**Reply by kallewoof** [reply]
322, not 344. If someone decides to do this definitely hit me up, yeah.

**Reply by ChristopherA** [reply]
:-/ fixed.

**Reply by deymow** [reply]
Ademola Adeyemi

**Reply by ChristopherA** [reply]
Chapter 9 in "Learning Bitcoin from the Command Line" teaches the basics of Bitcoin Script, and leverages btcdeb (the bitcoin debugger, also by @kallewoof) to test your scripts: https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/blob/master/09_3_Testing_a_Bitcoin_Script.md

The C++ source for [btcdeb](https://github.com/bitcoin-core/btcdeb) l would potentially be a good start for the creation of an offline BIP-322 message signing and verifying command-line app.

Long term, I'm also interested in if there is a subset of Bitcoin Script that might be more broadly useful as a form of [Smart Signatures](https://sourcecrypto.pub/posts/transcripts/smart-signatures-christopher-allen/), or if some specialized op_codes just for message signing would be a useful addition.

**Reply by ChristopherA** [reply]
@deymow wrote:
> "not an advanced coder thou. lool"

We always need more testers and documentation people!

**Reply by shoryak** [reply]
Interested to contribute in this

**Reply by shikharvashistha** [reply]
Interested

**Reply by Eunoia1729** [reply]
Interested to work on this!

**Reply by wip-abramson** [reply]
Hey, 
I am also happy to support/work with anyone taking this on. I have been exploring this bip for the past month or so on behalf of Digital Contract Design. I have developed a separate implementation using Jimmy Song's [buidl-python](https://github.com/buidl-bitcoin/buidl-python/) library. Currently, this exists in a set of Jupyter notebooks - https://github.com/LegReq/bip0322-signatures, which I would be happy to go over with people. I think it should be a useful tool for understanding how bip322 signatures work in practice - especially for those not well versed in c++.

At some point I plan to submit a P.R. to add this implementation to the buidl-python bitcoin library.

**Reply by aisha-ahmadi** [reply]
Interested! - Aisha

**Reply by rdeepu13** [reply]
Interested to work on this

**Reply by devesh-002** [reply]
Interested in this -devesh

**Reply by ahsmha** [reply]
Interested!
-ahmed

**Reply by shannona** [reply]
Logged as #112

**Comment by shannona** [comment]
## Project: Translate Learning Bitcoin from the Command Line

Last year saw two translations of Learning Bitcoin from the Command Line, into Spanish and Portuguese. We'd love to see more. Our goal is to reach large groups of engineers who might not have proficiency in one of our current course languages (English, Portuguese, or Spanish). Hindi? Italian? Chinese? We're happy to talk about options.

This is another project that can be led by a non-engineer, though experience with command-line UNIX/MacOS usage is a must. Also note that it _requires_ at least two people for any translation: one to translate a chapter, and the other to review it. (Those roles can switch about if desired.) Any reviewer must be a native speaker of the language or have a similar level of expertise.

Here's a document on the translation process, built on our experiences last year:
https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/blob/master/TRANSLATING.md

**Reply by shannona** [reply]
Logged as #126

**Comment by shannona** [comment]
## Project: Conquer Death with Bitcoin
A serious problem with any #SmartCustody solution in figuring out how to insure that your Bitcoin or other digital assets find your way to your heirs. Even if you're not dead, you might be incapacitated such that you can't personally access your funds, and it might be critically important for you to do so (say, to pay your ridiculous hospital bills in America).

We've tackled this problem twice:

* [Using TImelocks](https://github.com/BlockchainCommons/SmartCustody/blob/master/Docs/Timelocks.md) — An article we wrote on the state of timelocks, concluding that they weren't ready yet for usage due to a lack of integration in `bitcoin-cli` (or anywhere else)
* [mori-cli](https://github.com/BlockchainCommons/mori-cli) — An command-line program that used timelocks to protect wealth for heirs. I don't know the final state of the project.

It would be great to get this work to a more complete stage, where it could actually be used as a reference for how to protect digital assets for heirs. We think these timelocks are a great way to do so, but they don't have to be the only solution. So, a project might involve bringing `mori-cli` out of the alpha stage, or updating our #SmartCustody writing if the state of `bitcoin-cli` has improved, or offering more solutions, or possibly putting that all together into a menu for people to choose from.

**Reply by shoryak** [reply]
Interested in this but not familiar with rust :)

**Reply by bwakedavis** [reply]
Davis Bwake

**Reply by ChristopherA** [reply]
As we seem to have sufficient critical mass to begin ths project, I have started a community project issue to continue our discussion: https://github.com/BlockchainCommons/Community/issues/112 

I also have created a private Signal group — contact me for an invite.

**Reply by Eunoia1729** [reply]
Interested to collaborate on this

**Reply by devesh-002** [reply]
Interested!

**Reply by shannona** [reply]
#112 was actually the BIP-322 project. I've logged this as #127.

Generally, I'd say you _don't have_ to have Rust familiarity. Our core desire is to have a workable solution. We do have one past project that was started with Rust, but you could opt to continue with that, or to experiment with a new methodology.

**Comment by ChristopherA** [comment]
## Project: Update DIDs for secp256k1 as per BIP39

This would be a good group project for interns interested in internal standards and self-sovereign identity, and with different skills (writing specs, testing, coding javascript, coding C or rust, writing documentation, etc.)

Few of the emerging W3C DID (Decentralized Identifier) methods are using secp256k1 (the curve that bitcoin uses).  In addition, none of that I know are actively implementing Schnorr signatures for k1, and even those in spec-only form  are not making some of the choices that BIP40 made, or leverage Schnorr in sufficiently security reviewed libsecp256k1 library. Finally, no one in implementing DIDs currently are leveraging the multisig opportunities Schnorr, including musig2, FROST, adapter signatures, blinded signatures, etc.

In particular:

* Write up use cases for what secp256k1 can offer leveraging k1 multisig to DIDs and Verifiable Credentials.
* Suggest changes and test vectors for existing `did:key` [method](https://w3c-ccg.github.io/did-method-key/) to support k1 and Schnorr.
* Write reference code in javascript and other languages to transorm BIP340 keys into [multibase](https://datatracker.ietf.org/doc/html/draft-multiformats-multibase) & back, and register those with next update to the IETF internet-draft.
* Update DIF (Decentralized Identity Foundation) JSON-LD [SchnorrSecp256k1Signature2019](https://identity.foundation/SchnorrSecp256k1Signature2019/)https://github.com/decentralized-identity/SchnorrSecp256k1Signature2019) spec for BIP340 differences, and update test vectors.
* Update reference javascript library for JSON-LD SchnorrSecp256k1Signature2019 and investigate other implementations.
* Investigate JWT support of k1 Schnorr rather than ECDSA. Possibly submit IETF internet-draft or PR to JWT group to add k1 Schnorr.
* Work toward a CBOR-LD version of SchnorrSecp256k1Signature2019
* Create example DID objects and Verifiable Credentials leveraging SchnorrSecp256k1Signature2019
* Investigate Ristretto255 versions of the above.
* Present these at W3C CCG (Credentials Community Group) and DIF.

Related to this is beginning to be able to `did:btcr` 2.0 someday, see: 

* List of [BTCR v0 resources](https://github.com/w3c-ccg/didm-btcr/blob/gh-pages/resources.md)
* Podcast: [The Granddaddy of DIDs: BTCR](https://the-rubric.castos.com/episodes/the-granddaddy-of-dids-didbtcr)

Potential mentors include: @kimdhamilton @csuwildcat @jandrieu @rxgrant

**Reply by shoryak** [reply]
Interested  , Shorya Kumar

**Reply by shannona** [reply]
Issue #128

**Reply by shikharvashistha** [reply]
Interested

**Reply by rdeepu13** [reply]
Interested, Deepika

**Reply by tikka01** [reply]
I am interested in this project.

**Comment by ChristopherA** [comment]
## Project: Best Practices for Open Development of Secure Software

This is another project that would be good for a team with mixed skills, and does not require deeper software engineering experience (but at least one should have experience with build processes).

Various organizations (Linux Foundation, Google, etc.) have offered guidance as to the practices of security, supply chain, etc. , but also emphasize enterprise and OS supply chain use cases. There are also a number of automated tools (apps, GitHub actions, etc.) that can be used to audit on some of these.

However, many are not practical for smaller projects, especially the emerging blockchain security repos, where only a few people may be contributing.

The goal of this project is to survey the existing recommended practices, best practices of various important security projects (including Blockchain Commons practices), etc., to identify which address the biggest threats given the effort (threat analysis), are practical for small projects to implement, which we might be able to offer some documentation and examples of how best to install and use, and guidance to contributors to small projects on how to tool and support this practices (like docs teaching git signing for writers contributing documentation to a secure repo).

**Reply by shannona** [reply]
Issue #129

**Comment by ChristopherA** [comment]
## Project: Secure OS Guides & Best Practices (Tails, Whonix, Qubes)

The war in Ukraine, and other authoritarian threats to citizens around the world demonstrate that we still need powerful tools for anonymity. One of the best of these is Tails OS, which runs on cheap computers from a USB device. Whonix runs on multiple virtual machines inside of your existing OS, and Qubes offers a very different security model on some specific laptops. All, however, are not easy to install or use by activists.

Interns from 2011 outlined some thoughts on documentation and scripts to make it easier to use Tails OS: https://github.com/BlockchainCommons/Community/discussions/43

**Reply by shannona** [reply]
New project labeled as #130

**Reply by ahsmha** [reply]
Interested in this!
-ahmed

**Comment by ChristopherA** [comment]
## Project: Survey of Cryptographic Auctions

With the rise of digital assets, there is an increasing need for fair pricing and exchange between them. 

For those of a more a cryptographic or mathematical bent, but are not software engineers, I'd love to see a survey of different auction types that now can be supported securely with modern cryptography. @shannona and I have a list of historical auction types and their utility for supporting different kinds of needs for auctions in different marketplaces, but many do not map to cryptographic solutions.  We'd like to offer the academic cryptographer community a target list of auction types NOT currently supported, to encourage innovation.

Goal of this group would be to submit the survey as a peer-reviewed paper to an academic journal or conference, or just public on Arxiv.org and make it available to all.

**Reply by shannona** [reply]
Issue #131
